metabase/metabase
# Vertical legend fixes for LineAreaBarChart

Fixes issues found during bugbash:
1. Fixes trimming in horizontal mode
2. Aligns the LineAreaBarChart in a special way in the Query Builder
3. Fixes vertical legend calculations for fullscreen mode
4. Fixes legends text color in night mode in dashboards

How to test 1:
- Add a line chart to a dashboard
- Edit dashboard -> hover over card -> Add series -> Add several other line charts
- When series names overflow the width of the card, they should be truncated

How to test 2:
- See @mazameli comment. It should be aligned correctly for both horizontal and vertical modes

How to test 3:
- Create a line chart with breakout by category, e.g. by User -> State
- Add this line chart to a dashboard
- Enter fullscreen mode
- You should be able to see "And X more" button, which should allow you to select series in the popover

How to test 4:
- Same as 3
- Enter night mode
- Legend labels should change their color
